src/main/java/org/candlepin/dto/api/v1/ConsumerTranslator.java

```diff
@@ -137,8 +138,16 @@ public ConsumerDTO populate(ModelTranslator translator, Consumer source, Consume
                 dest.setOwner(owner != null ? translator.translate(owner, NestedOwnerDTO.class) : null);
             }
 
-            Environment environment = this.environmentCurator.getConsumerEnvironment(source);
-            dest.setEnvironment(translator.translate(environment, EnvironmentDTO.class));
+            if (source.getEnvironmentIds() != null) {
+                Collection<String> envIDs = source.getEnvironmentIds().values();
+                for (String id : envIDs) {
+                    dest.addEnvironments(translator
+                        .translate(this.environmentCurator.listById(id), EnvironmentDTO.class));
```

Comment:
Instead of iterating over the environment ids here, we should instead create a new method `environmentCurator.getConsumerEnvironments`  (similar to `environmentCurator.getConsumerEnvironment`), which accepts a consumer object and returns the list of enviornments they are in, and then use that new curator method here

Reply:
Done.